tests/test_when.py

```diff
@@ -867,3 +867,33 @@ class _Test(m.Circuit):
 
     m.compile(f"build/{_Test.name}", _Test, output="mlir")
     assert check_gold(__file__, f"{_Test.name}.mlir")
+
+
+def test_when_latch_zext():
```

Comment:
also let's give this a name that describes what the test is actually testing (latch_nested_multiple ?)